osquery/database/plugins/ephemeral.cpp

```diff
@@ -16,10 +18,36 @@ Status EphemeralDatabasePlugin::get(const std::string& domain,
                                     const std::string& key,
                                     std::string& value) const {
   if (db_.count(domain) > 0 && db_.at(domain).count(key) > 0) {
-    value = db_.at(domain).at(key);
+    try {
+      value = boost::get<std::string>(db_.at(domain).at(key));
+    } catch (const boost::bad_get& e) {
+      LOG(WARNING) << "Type error getting string value for (domain,key) : ("
+                   << key << "," << domain << ") " << e.what();
+      return Status(
+          1,
+          "EphemeralDatabasePlugin::get was requested incorrect type(string)");
+    }
+    return Status(0);
+  } else {
+    return Status(1, "Key or domain does not exist");
+  }
+}
+
+Status EphemeralDatabasePlugin::get(const std::string& domain,
+                                    const std::string& key,
+                                    int& value) const {
+  if (db_.count(domain) > 0 && db_.at(domain).count(key) > 0) {
+    try {
+      value = boost::get<int>(db_.at(domain).at(key));
+    } catch (const boost::bad_get& e) {
+      LOG(WARNING) << "Type error getting int value for (domain,key) : (" << key
+                   << "," << domain << ") " << e.what();
+      return Status(
+          1, "EphemeralDatabasePlugin::get was requested incorrect type(int)");
+    }
     return Status(0);
   } else {
-    return Status(1);
+    return Status(1, "Key or domain does not exist");
```

Comment:
It is good to know in stack-trace which `key` and `domain` does not exist. Let's print it in `Status` error message.